dhcpv6/option_bootfileparam.go

```diff
@@ -0,0 +1,68 @@
+package dhcpv6
+
+import (
+	"fmt"
+
+	"github.com/u-root/u-root/pkg/uio"
+)
+
+// OptBootFileParam implements the OptionBootfileParam option
+//
+// This module defines the OPT_BOOTFILE_PARAM structure.
+// https://www.ietf.org/rfc/rfc5970.txt (section 3.2)
+type OptBootFileParam []string
+
+var _ Option = OptBootFileParam(nil)
+
+// Code returns the option code
+func (op OptBootFileParam) Code() OptionCode {
+	return OptionBootfileParam
+}
+
+// ToBytes serializes the option and returns it as a sequence of bytes
+func (op OptBootFileParam) ToBytes() []byte {
+	buf := uio.NewBigEndianBuffer(nil)
+	for _, param := range op {
+		if len(param) >= 1<<16 {
+			panic("a too long parameter")
+		}
+		buf.Write16(uint16(len(param)))
+		buf.WriteBytes([]byte(param))
+		/*if err := buf.Error(); err != nil {
+			// TODO: description of `WriteBytes` says it could return
+			// an error via `buf.Error()`. But a quick look into implementation of
+			// `WriteBytes` at the moment of this comment showed it does not set any
+			// errors to `Error()` output. It's required to make a decision:
+			// to fix `WriteBytes`'s description or to find a way to handle an error here.
+		}*/
+	}
+	return buf.Data()
+}
+
+func (op OptBootFileParam) String() string {
+	return fmt.Sprintf("OptBootFileParam(%v)", ([]string)(op))
+}
+
+// ParseOptBootFileParam builds an OptBootFileParam structure from a sequence
+// of bytes. The input data does not include option code and length bytes.
+func ParseOptBootFileParam(data []byte) (result OptBootFileParam, err error) {
+	buf := uio.NewBigEndianBuffer(data)
+	for buf.Has(2) {
+		length := buf.Read16()
+		if err := buf.Error(); err != nil {
+			return nil, fmt.Errorf("[bootfile param] unable to read the parameter length: %w", err)
+		}
+
+		param, err := buf.ReadN(int(length))
```

Comment:
I think you want `buf.CopyN` here instead, which populates `Error()` like the other `Read*` functions. `ReadN` is more of a low-level functions and gives you a direct slice into the underlying buffer (I do think the name ambiguity between the uio.Buffer and uio.Lexer functions is misleading)

This doesn't matter too much since you convert to string anyway, but it would allow you to drop the error check as well.

Reply:
> I think you want `buf.CopyN` here instead

Yeah, such a method was intended initially. And yes, I mistook because of this:

> I do think the name ambiguity between the uio.Buffer and uio.Lexer functions is misleading

However ReadN isn't worse is this case, because (as you already said) I convert it to string, anyway. And ReadN even allows to avoid an extra memory allocation and copy. And the error check will be required anyway -- I should not add `string([]byte(nil))` to `result`. On the other hand a next programmer may think it returns a copy (because the name looks like other `Read*` functions) and make a mistake. I can change it if required, should I?

Reply:
you're right about the confusion, it's weird. do you have a suggestion?

I think here you should use CopyN -- it makes a copy, but it avoids the unnecessary error check.

Reply:
(there should probably be a ReadN that doesn't return an error and folds it up to FinError)